test/spec/services/apiServiceSpec.js

```diff
@@ -118,20 +118,57 @@ describe("APIService", function() {
       ]
     ];
     _.each(rgvs, _.spread(function(rgv, expectedAPIInfo) {
-      it('should result in ' + JSON.stringify(expectedAPIInfo) + ' when called with ' + JSON.stringify(rgv), function() {
-        var actualAPIInfo = APIService.apiInfo(rgv);
-        if(actualAPIInfo) {
-          expect(actualAPIInfo.resource).toEqual(expectedAPIInfo.resource);
-          expect(actualAPIInfo.group).toEqual(expectedAPIInfo.group);
-          expect(actualAPIInfo.version).toEqual(expectedAPIInfo.version);
-          expect(actualAPIInfo.hostPort).toEqual(expectedAPIInfo.hostPort);
-          expect(actualAPIInfo.prefix).toEqual(expectedAPIInfo.prefix);
-          expect(actualAPIInfo.namespaced).toEqual(expectedAPIInfo.namespaced);
-          expect(actualAPIInfo.verbs).toEqual(expectedAPIInfo.verbs);
-        } else {
-          expect(actualAPIInfo).toEqual(expectedAPIInfo);
-        }
-      });
+        it('should return the correct resource', function() {
+          var actualAPIInfo = APIService.apiInfo(rgv);
+          if(actualAPIInfo) {
+            expect(actualAPIInfo.resource).toEqual(expectedAPIInfo.resource);
+          }
+        });
+        it('should return the correct group', function() {
+          var actualAPIInfo = APIService.apiInfo(rgv);
+          if(actualAPIInfo) {
+            expect(actualAPIInfo.group).toEqual(expectedAPIInfo.group);
+          }
+        });
+        it('should return the correct version', function() {
+          var actualAPIInfo = APIService.apiInfo(rgv);
+          if(actualAPIInfo) {
+            expect(actualAPIInfo.version).toEqual(expectedAPIInfo.version);
+          }
+        });
+        it('should return the correct hostPort', function() {
+          var actualAPIInfo = APIService.apiInfo(rgv);
+          if(actualAPIInfo) {
+            expect(actualAPIInfo.hostPort).toEqual(expectedAPIInfo.hostPort);
+          }
+        });
+        it('should return the correct prefix', function() {
+          var actualAPIInfo = APIService.apiInfo(rgv);
+          if(actualAPIInfo) {
+            expect(actualAPIInfo.prefix).toEqual(expectedAPIInfo.prefix);
+          }
+        });
+        it('should return the correct namespaced', function() {
+          var actualAPIInfo = APIService.apiInfo(rgv);
+          if(actualAPIInfo) {
+            expect(actualAPIInfo.namespaced).toEqual(expectedAPIInfo.namespaced);
+          }
+        });
+        it('should return the correct verbs', function() {
+          var actualAPIInfo = APIService.apiInfo(rgv);
+          if(actualAPIInfo) {
+            expect(actualAPIInfo.verbs).toEqual(expectedAPIInfo.verbs);
+          }
+        });
+        // TODO: Is this valid or a bad test?
+        // I don't have any idea why it would be undefined.
+        // If its correct, we should keep it and clarify the condition.
+        // it('should be undefined for an unknown reason', function() {
```

Comment:
The case at L116
I believe the point is that its bogus api information that won't resolve to an actual type, so the call to get the api info should return undefined/null

Reply:
ok makes sense.  updated.